src/js/relay.firefox.com/installation_indicator.js

```diff
@@ -4,8 +4,27 @@
 
 (async () => {
   localStorage.setItem("fxRelayAddonInstalled", "true");
-  document.querySelectorAll("firefox-private-relay-addon").forEach((el) => {
+  document.querySelectorAll("firefox-private-relay-addon").forEach(async (el) => {
     el.dataset.addonInstalled = "true";
+
+    // In the server-rendered version of the website, the add-on would store alias labels
+    // locally if server-side storage was disabled.
+    // In the React version of the website, the website handles local storage itself.
```

Comment:
I don't follow this ... regardless of whether the website is React or not, we still want the add-on to store labels locally if server-side storage is disabled.

Reply:
I went with a slightly different approach in the React website (already discussed with @maxxcrawford): now, the website instead of the add-on stores the labels locally (i.e. in localStorage). It only does this if the add-on is installed and server-side storage is enabled.

The main advantage of this is that we have a single code path for editing labels, which just at the very end determines whether to store them locally or on the server. This avoids the situation we have now, where the add-on just reaches into the website DOM and manipulates it, which:

- means that changing the website can break the add-on's interaction with it
- results in different label behaviour with local vs server-side storage (e.g. there's some validation that's only applied in local storage today)
- requires add-on deployments to make changes

In this new approach, the add-on injects the existing labels, which the website then imports once, and from then on it handles local storage of labels.

Reply:
Oh, interesting. Sorry to throw a wrench in that plan, but what happens when someone with Relay extension or add-on uses "Clear site data" in their browser? That will remove all their alias labels? Even though they still have the Relay extension installed?

Reply:
Yes, that would also remove the labels. That said, that would match expected behaviour for me, just like how I'm logged out of the Relay website when I do that.

In fact, it would even make sense to me to take this one step further: we can locally store alias labels even _without_ the add-on installed. The way I've currently implemented this, is to have an explicit check for if the add-on is installed, and if not, to just not display labels at all, so this could be achieved by simply removing that check.

Reply:
That's interesting to do this without the add-on all together. 

Question: If a user clears site data right now, does that clear the add-on stored labels? If that's the case, we would need to either: 
- Provide a way to back up/export/import labels locally for private-conscious users
- Provide notice of this change before we actually implement it.

Reply:
I just cleared all history:

![image](https://user-images.githubusercontent.com/4251/154250125-632da925-4c2e-4c7f-b1fe-f2750f078e8f.png)

That does not clear the stored labels. I would think the group of people who

- Have disabled server-side label storage, and
- Have locally-stored labels, and
- Regularly clear site data, and
- Expect that to not remove their locally stored labels, which they primarily see in the website, and
- Would go through the hassle of exporting and importing their labels every time they do that

is in the low double digits, if not single digits (or even 0!), so building a custom export/import system seems like not worth the cost.

Adding a notice is less effort, but even if we only show that to people who have locally stored labels, I'd be a bit worried that the vast majority of them would never experience any issue with this, and thus such a warning might only serve to confuse them. But that's something we might want to bring up with Tony and Chris.

Reply:
OK, so I just spent some time moving label storage back into the add-on. In addition to the concern above, a more pressing issue was that the add-on cannot use labels in e.g. context menus if they are stored in the website context.

Thus, I added 1df7cdd3. This works in cooperation with https://github.com/mozilla/fx-private-relay/pull/1518/commits/ab545456a1b8571e43ccde4253a646558de35125 and https://github.com/mozilla/fx-private-relay/pull/1518/commits/1df7cdd3de3a5b795cdee684940e7b1f32528622: when a label is updated on the website, it now triggers an event on the `<firefox-private-relay-addon>` element. The add-on listens for those events and updates its local cache of aliases. This preserves the fact that the add-on is no longer reaching into the website DOM, resulting in inconsistent flows, and sets a clear API boundary (the custom element) for the website and add-on to communicate through, while moving the label and `generated_for` data into the add-on.

Reply:
Got'cha. I appreciate the pivot!

Reply:
Neat trick! @Vinnl and I discussed a bit today; this implementation seems safe enough - @Vinnl pointed out that this content script only runs on our own profile page.